21 allow user specific sessions

NicolasSeay opened the pull request:
Allows for user-specific sessions. Closing the tab will end the session.
Added registration error message.
No longer base login error message on userId of 0, moved it to app state along with registration error.

jmhohe commented on MealPlanner/src/main/java/com/nico/mp/controllers/RecipeController.java:
```diff
@@ -18,28 +20,54 @@ public class RecipeController {
+	) {
 		log.info("Recipes - received recipe list request");
-		List<Recipe> recipes = recipeService.getAllRecipes(userId);
+		if (authorization == null || !jwtUtil.validateToken(userId, authorization.substring(7))) {
```
Don't really need the authorization == null here

NicolasSeay commented:
Good catch, I refactored how we handle backend authentication and haven't
updated all the controllers accordingly

On Wed, Aug 16, 2023, 10:00 PM jmhohe ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In MealPlanner/src/main/java/com/nico/mp/controllers/RecipeController.java
> <https://github.com/NicolasSeay/MealPlanner/pull/22#discussion_r1296596030>
> :
>
> >  		log.info("Recipes - received recipe list request");
> -		List<Recipe> recipes = recipeService.getAllRecipes(userId);
> +		if (authorization == null || !jwtUtil.validateToken(userId, authorization.substring(7))) {
>
> Don't really need the authorization == null here
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NicolasSeay/MealPlanner/pull/22#pullrequestreview-1581651405>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADY6MMEFYBR4LYEKFL2JTTLXVV3MTANCNFSM6AAAAAA3QN3QCA>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>
